pygmt/src/inset.py

```diff
@@ -0,0 +1,128 @@
+        in plot coordinates (inches, cm, etc.). By default, the anchor
+        point on the scale is assumed to be the bottom left corner (BL),
+        but this can be changed by appending **+j** followed by a 2-char
+        justification code *justify*.
+        **Note**: If **j** is used then *justify* defaults to the same
+        as *refpoint*, if **J** is used then *justify* defaults to the
+        mirror opposite of *refpoint*. Specify inset box attributes via
+        the ``box`` option [outline only].
+    box : str or bool
+        [**+c**\ *clearances*][**+g**\ *fill*][**+i**\ [[*gap*/]\
+        *pen*]][**+p**\ [*pen*]][**+r**\ [*radius*]][**+s**\
+        [[*dx*/*dy*/][*shade*]]]
+
+        If passed ``True``, this draws a rectangular box around the map
+        inset using the default pen; specify a different pen
+        with **+p**\ *pen*. Add **+g**\ *fill* to fill the logo box
+        [Default is no fill].
+        Append **+c**\ *clearance*  where *clearance* is either
+        *gap*, *xgap*\ /\ *ygap*, or *lgap*\ /\ *rgap*\ /\ *bgap*\ /\
+        *tgap* where these items are uniform, separate in x- and
+        y-direction, or individual side spacings between logo and border.
+        Append **+i** to draw a secondary, inner border as well. We use a
+        uniform *gap* between borders of 2\ **p** and the default pen
+        unless other values are specified. Append **+r** to draw rounded
+        rectangular borders instead, with a 6\ **p** corner radius. You
+        can override this radius by appending another value. Append
+        **+s** to draw an offset background shaded region. Here, *dx*/*dy*
+        indicates the shift relative to the foreground frame
+        [4\ **p**/-4\ **p**] and *shade* sets the fill style to use for
+        shading [Default is gray50].
+    margin : int or str or list
+        This is clearance that is added around the inside of the inset.
+        Plotting will take place within the inner region only. The margins
+        can be a single value, a pair of values separated (for setting
+        separate horizontal and vertical margins), or the full set of four
+        margins (for setting separate left, right, bottom, and top
+        margins). When passing multiple values, it can be either a list or
+        a string with the values separated by forward
+        slashes [Default is no margins].
+    {V}
+    Examples
+    --------
+    >>> # Create the larger figure
+    >>> fig.coast(
+    ...     region="MG+r2",
+    ...     land="brown",
+    ...     water="lightblue",
+    ...     shorelines="thin",
+    ...     frame="a",
+    ... )
+    >>> # Create an inset, setting the position to top left, the width to 3.5 centimeters, and
+    >>> # Use a "with" statement to use the inset context manager
+    >>> with fig.inset(position="jTL+w3.5c+o0.2c", margin=0, box="+pgreen"):
+    ...     # Map elements plotted under the "with" statement are added to the inset
+    ...     fig.coast(
```

Comment:
```suggestion
    {V}

    Examples
    --------
    >>> fig = pygmt.Figure()  # Initialize a figure first
    >>> # Use a "with" statement to initialize the inset context manager
    >>> # Setting the position to top left and a width of 3.5 centimeters
    >>> with fig.inset(position="jTL+w3.5c+o0.2c", margin=0, box="+pgreen"):
    ...     # Map elements under the "with" statement are ploted in the inset
    ...     fig.coast(
```

Need a blank line under `{V}`, and it's probably enough to just use `fig = pygmt.Figure()` here and just show how `with fig.inset()` works.